internal/webhook/external/server.go

```diff
@@ -71,3 +68,71 @@ func (s *server) Serve(ctx context.Context, l net.Listener) error {
 		return err
 	}
 }
+
+func (s *server) route(w http.ResponseWriter, r *http.Request) {
+	ctx := (r.Context())
+	logger := logging.LoggerFromContext(ctx)
+	logger.Debug("refresh route handler called", "path", r.URL.Path)
+	var projectConfigs kargoapi.ProjectConfigList
+	err := s.client.List(
+		ctx,
+		&projectConfigs,
+		client.MatchingFields{
+			indexer.ProjectConfigsByWebhookReceiverPathsField: r.URL.Path,
+		},
+	)
+	if err != nil {
+		logger.Error(err, "failed to list project configs")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(projectConfigs.Items) == 0 {
+		logger.Info("no project configs found")
+		http.Error(w, "no project configs found for the request", http.StatusNotFound)
+		return
+	}
+
+	// Projects are allowed to have, at most, a single ProjectConfig
+	pc := projectConfigs.Items[0]
+	logger.Info("found project config", "project-config", pc)
+	wrc, err := s.getWebhookReceiverConfig(pc, r.URL.Path)
+	if err != nil {
+		logger.Error(err, "failed to find webhook receiver config")
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	logger.Debug("webhook receiver config found", "webhook-receiver-config", wrc)
+	switch {
+	case wrc.GitHub != nil:
+		githubHandler(
+			s.client,
+			pc.Namespace,
+			wrc.GitHub.SecretRef.Name,
+		)(w, r)
+	default:
+		http.Error(w, "not found", http.StatusNotFound)
+	}
+}
+
+func (s *server) getWebhookReceiverConfig(
+	pc kargoapi.ProjectConfig,
+	receiverPath string,
+) (*kargoapi.WebhookReceiverConfig, error) {
+	var whrc *kargoapi.WebhookReceiverConfig
+	for i, r := range pc.Status.WebhookReceivers {
+		if receiverPath == r.Path {
+			whrc = &pc.Spec.WebhookReceivers[i]
```

Comment:
I know we have receiver config and receivers names. Weren't we going to match on those instead of assuming the two lists are ordered the same? (Assuming/requiring parallel lists to be ordered the same widely regarded as rather brittle, so we avoid it as a rule.)